keystore/v2/keystore/api/tests/keyStore.go

```diff
@@ -19,9 +19,110 @@ package tests
 
 import (
 	"testing"
+	"time"
 
 	"github.com/cossacklabs/acra/keystore/v2/keystore/api"
+	"github.com/cossacklabs/acra/keystore/v2/keystore/crypto"
 )
 
 // NewKeyStore is a factory of KeyStore under testing.
 type NewKeyStore func(t *testing.T) api.MutableKeyStore
+
+const (
+	exportRingKeyPair   = "keyring/keypair"
+	exportRingPublic    = "keyring/public"
+	exportRingSymmetric = "keyring/symmetric"
+)
+
+var exportRingAll = []string{
+	exportRingKeyPair,
+	exportRingPublic,
+	exportRingSymmetric,
+}
+
+// TestKeyStore runs KeyStore test suite.
+func TestKeyStore(t *testing.T, newKeyStore NewKeyStore) {
+	t.Run("TestKeyStoreExport", func(t *testing.T) {
+		testKeyStoreExport(t, newKeyStore)
+	})
+}
+
+var (
+	testExportMasterKey    = []byte("test export master key")
+	testExportSignatureKey = []byte("test export signature key")
+)
+
+func newExportStoreSuite(t *testing.T) *crypto.KeyStoreSuite {
+	encryptor, err := crypto.NewSCellSuite(testExportMasterKey, testExportSignatureKey)
+	if err != nil {
+		t.Fatalf("cannot create encryptor: %v", err)
+	}
+	return encryptor
+}
+
+func setupDemoKeyStore(s api.MutableKeyStore, t *testing.T) {
+	ringKeyPair, err := s.OpenKeyRingRW(exportRingKeyPair)
+	if err != nil {
+		t.Errorf("failed to create key ring: %v", err)
+	}
+	_, err = ringKeyPair.AddKey(api.KeyDescription{
+		ValidSince: time.Now(),
+		ValidUntil: time.Now().Add(time.Hour),
+		Data: []api.KeyData{
+			api.KeyData{
+				Format:     api.ThemisKeyPairFormat,
+				PublicKey:  []byte("public key"),
+				PrivateKey: []byte("private key"),
+			},
+		},
+	})
+	if err != nil {
+		t.Errorf("failed to add key pair: %v", err)
+	}
+
+	ringPublic, err := s.OpenKeyRingRW(exportRingPublic)
+	if err != nil {
+		t.Errorf("failed to create key ring: %v", err)
+	}
+	_, err = ringPublic.AddKey(api.KeyDescription{
+		ValidSince: time.Now(),
+		ValidUntil: time.Now().Add(time.Hour),
+		Data: []api.KeyData{
+			api.KeyData{
+				Format:    api.ThemisKeyPairFormat,
+				PublicKey: []byte("only public key"),
+			},
+		},
+	})
+	if err != nil {
+		t.Errorf("failed to add public key: %v", err)
+	}
+
+	ringSymmetric, err := s.OpenKeyRingRW(exportRingSymmetric)
+	if err != nil {
+		t.Errorf("failed to create key ring: %v", err)
+	}
+	_, err = ringSymmetric.AddKey(api.KeyDescription{
+		ValidSince: time.Now(),
+		ValidUntil: time.Now().Add(time.Hour),
+		Data: []api.KeyData{
+			api.KeyData{
+				Format:       api.ThemisSymmetricKeyFormat,
+				SymmetricKey: []byte("symmetric key"),
+			},
+		},
+	})
+	if err != nil {
+		t.Errorf("failed to add public key: %v", err)
+	}
+}
+
+func testKeyStoreExport(t *testing.T, newKeyStore NewKeyStore) {
+	s := newKeyStore(t)
+	setupDemoKeyStore(s, t)
+
+	_, err := s.ExportKeyRings(exportRingAll, newExportStoreSuite(t))
```

Comment:
> why not check that exported data is the same as expected?

It’s all encrypted so it’s hard to verify the contents. Right now this test is more of a placeholder to check that you can export stuff without an error. It will be properly verified in the next PR that adds import. Then we’d try importing the exported data and check that we get the key data we should get.